aries_vcx/src/common/proofs/verifier/verifier.rs

```diff
@@ -135,28 +137,31 @@ pub mod unit_tests {
     #[ignore]
     async fn test_pool_proof_restrictions() {
         SetupProfile::run(|setup| async move {
-            let requested_attrs = json!([
-                json!({
-                    "name":"address1",
-                    "restrictions": [{ "issuer_did": "Not Here" }],
-                }),
-                json!({
-                    "name":"zip",
-                }),
-                json!({
-                    "name":"self_attest",
-                    "self_attest_allowed": true,
-                }),
-            ])
-            .to_string();
+            let requested_attrs = vec![
+                AttrInfo {
+                    name: Some(String::from("address1")),
+                    restrictions: Some(Restrictions::V2(json!([{ "issuer_did": "Not Here" }]))),
+                    ..AttrInfo::default()
+                },
+                AttrInfo {
+                    name: Some(String::from("zip")),
+                    ..AttrInfo::default()
+                },
+                AttrInfo {
+                    name: Some(String::from("self_attest")),
+                    self_attest_allowed: Some(true),
+                    ..AttrInfo::default()
+                },
+            ];
+
```

Comment:
@Patrik-Stas did you mean serializing or this? This falls short in case of refactoring some tests which depend on constant `AttrInfo`.

Reply:
I am not sure what you mean, but as I mentioned before, you need to eliminate usage of 
`set_requested_attributes_as_string/ set_requested_predicates_as_string` in aries-vcx

For example this test
```
    #[tokio::test]
    async fn test_should_return_error_if_name_and_names_passed_together() {
        let _setup = SetupDefaults::init();

        let attr_info = json!({
          "name": "name",
          "names": ["name", "age", "email"],
          "restrictions": [{"schema_id": "6XFh8yBzrpJQmNyZzgoTqB:2:schema_name:0.0.11"}]
        });

        let requested_attrs = json!([attr_info]).to_string();

        let anoncreds: Arc<dyn BaseAnonCreds> = Arc::new(MockAnoncreds {});
        let err = ProofRequestData::create(&anoncreds, "")
            .await
            .unwrap()
            .set_requested_attributes_as_string(requested_attrs.into())
            .unwrap_err();

        assert_eq!(AriesVcxErrorKind::InvalidProofRequest, err.kind());
    }
```
adjust like this

```
    #[tokio::test]
    async fn test_should_return_error_if_name_and_names_passed_together() {
        let _setup = SetupDefaults::init();

        let requested = vec!(AttrInfo {
            name: "name".into(),
            names: Some(vec!("name".to_string(), "age".to_string(), "email".to_string())),
            restrictions: Some(Restrictions::V1(vec!(Filter{
                schema_id: Some("6XFh8yBzrpJQmNyZzgoTqB:2:schema_name:0.0.11".to_string()),
                schema_issuer_did: None,
                schema_name: None,
                schema_version: None,
                issuer_did: None,
                cred_def_id: None,
            }))),
            non_revoked: None,
            self_attest_allowed: None,
        });

        let requested_attrs = json!([attr_info]).to_string();

        let anoncreds: Arc<dyn BaseAnonCreds> = Arc::new(MockAnoncreds {});
        let err = ProofRequestData::create(&anoncreds, "")
            .await
            .unwrap()
            .set_requested_attributes_as_vec(requested)
            .unwrap_err();

        assert_eq!(AriesVcxErrorKind::InvalidProofRequest, err.kind());
    }
```
So we directly construct typed list of requested attributes, rather than passing string and then parsing that.

The downside which emerged as I wrote this example is that the construction of `restrictions` value especially is quite cumbersome. It would be nice to apple builder pattern for constructing these structures to make them more compact. Are you familiar with builder pattern? (it's not rust specific thing, just in context of programming in general)

Given this, I would update the plan to:
- create builder pattern for `Filter` structure
- continue with original advice to eliminate usage of `set_requested_attributes_as_string/ set_requested_predicates_as_string`
- then once `set_requested_attributes_as_string/ set_requested_predicates_as_string` become dead code in aries-vcx, move the code to libvcx - it will be trivial at that point because no code in `aries-vcx`  would be using it anymore.